### PR TITLE
Add in audit review recurrence presets.

### PR DESCRIPTION
**Pull request by mdwn**
Access Lists now have audit review recurrence presets. These allow users to specify review frequencies of 1, 3, 6, or 12 months, and specify the 1st, 15th, or last days of the target month. Presets have been used for their simplicity over other various recurrence definition mechanisms, as these presets are much clearer than many of the other options.

### CODE REVIEW

File: api/types/accesslist/accesslist.go [hunk 1]
```diff
@@ -157,11 +253,53 @@ func (a *AccessList) CheckAndSetDefaults() error {
...
 
-	if a.Spec.Audit.Frequency == 0 {
-		return trace.BadParameter("audit frequency must be greater than 0")
+	// Frequency should be converted into a recurrence.
+	if a.Spec.Audit.Frequency != 0 {
+		if a.Spec.Audit.Recurrence.Frequency == 0 && a.Spec.Audit.Recurrence.DayOfMonth == 0 {
+			// Just assume the first of the month.
+			a.Spec.Audit.Recurrence.DayOfMonth = FirstDayOfMonth
+
+			numMonths := int(a.Spec.Audit.Frequency / averageNumberOfHoursInAMonth)
```
Comment (r0mant) [inline api/types/accesslist/accesslist.go:262]: Not sure I understand, if we're doing specific presets, why do we need to calculate anything?

Comment (mdwn) [reply api/types/accesslist/accesslist.go:262]: This is trying to take the existing frequency field and converting it to a preset. Basically trying to be backwards compatible here.

Comment (r0mant) [reply api/types/accesslist/accesslist.go:262]: Do we need to be backwards compatible? Can we deprecate the old field?

**Review by r0mant — APPROVED** [review]

File: api/types/accesslist/accesslist.go [hunk 2]
```diff
@@ -82,11 +162,20 @@ type Owner struct {
...
+	// Recurrence is the recurrence definition for auditing.
+	Recurrence Recurrence `json:"recurrence" yaml:"recurrence"`
+}
+
+// Recurrence defines when access list reviews should occur.
+type Recurrence struct {
+	// Frequency is the frequency between access list reviews.
+	Frequency ReviewFrequency `json:"frequency" yaml:"frequency"`
+
+	// DayOfMonth is the day of month subsequent reviews will be scheduled on.
```
Comment (jakule) [inline api/types/accesslist/accesslist.go:177]: Can you add that the allowed values are `1, first, 15, last`?

**Review by jakule — APPROVED** [review]
One nit

File: api/proto/teleport/accesslist/v1/accesslist.proto [hunk 3]
```diff
@@ -80,11 +79,40 @@ message AccessListOwner {
 
 // AccessListAudit describes the audit configuration for an access list.
 message AccessListAudit {
-  // frequency is a duration that describes how often an access list must be audited.
-  google.protobuf.Duration frequency = 1;
```
[the remark this answers is not in the document]
Comment (mdwn) [reply api/proto/teleport/accesslist/v1/accesslist.proto:84]: You're not wrong. I was originally intending on keeping the `frequency` around and having it convert, but I got some pushback on that end. I have (and have had) a fix for this: https://github.com/gravitational/teleport.e/pull/2324, but it's still waiting on reviews.

**Pull request merged**